vmm/src/lib.rs

```diff
@@ -582,15 +582,16 @@ impl Vmm {
         }
 
         // First we stop the current VM and create a new one.
-        if let Some(ref mut vm) = self.vm {
+        if let Some(mut vm) = self.vm.take() {
             let config = vm.get_config();
             let serial_pty = vm.serial_pty();
             let console_pty = vm.console_pty();
             let console_resize_pipe = vm
                 .console_resize_pipe()
                 .as_ref()
                 .map(|pipe| pipe.try_clone().unwrap());
-            self.vm_shutdown()?;
+            vm.shutdown()?;
+            std::mem::drop(vm);
```

Comment:
I think calling drop explicitly is a bad idea. The `.take()` will be sufficient to lead to the drop when the block is closed. Perhaps if you're worried about side effects you could close this block immediately after the `vm.shutdown()`.

Reply:
I'm okay with the alternative you're proposing :)